test/widgets/compose_box_test.dart

```diff
@@ -0,0 +1,106 @@
+      //   but a bit different -- could maybe deduplicate some.
+      parseMarkedText(String markedText) {
+        final textBuffer = StringBuffer();
+        int? insertionPoint;
+        int i = 0;
+        for (final char in markedText.codeUnits) {
+          if (char == 94 /* ^ */) {
+            if (insertionPoint != null) {
+              throw Exception('Test error: too many ^ in input');
+            }
+            insertionPoint = i;
+            continue;
+          }
+          textBuffer.writeCharCode(char);
+          i++;
+        }
+        if (insertionPoint == null) {
+          throw Exception('Test error: expected ^ in input');
+        }
+        return TextEditingValue(text: textBuffer.toString(), selection: TextSelection.collapsed(offset: insertionPoint));
+      }
+
+      /// Test the given `insertPadded` call, in a convenient format.
+      ///
+      /// In valueBefore, represent the insertion point as "^".
+      /// In expectedValue, represent the collapsed selection as "^".
+      checkInsertPadded(String description, String valueBefore, String textToInsert, String expectedValue) {
+        test(description, () {
+          final controller = ComposeContentController();
+          controller.value = parseMarkedText(valueBefore);
+          controller.insertPadded(textToInsert);
+          check(controller.value).equals(parseMarkedText(expectedValue));
+        });
+      }
+
+      // TODO(?) exercise the part of insertPadded that chooses the insertion
+      //   point based on [TextEditingValue.selection], which may be collapsed,
+      //   expanded, or null (what they call !TextSelection.isValid).
+
+      checkInsertPadded('empty; insert one line',
+        '^', 'a\n',    'a\n\n^');
+      checkInsertPadded('empty; insert two lines',
+        '^', 'a\nb\n', 'a\nb\n\n^');
+
+      group('insert at end', () {
+        checkInsertPadded('one empty line; insert one line',
+          '\n^',     'a\n',    '\na\n\n^');
+        checkInsertPadded('two empty lines; insert one line',
+          '\n\n^',   'a\n',    '\n\na\n\n^');
+        checkInsertPadded('one line, incomplete; insert one line',
+          'a^',      'b\n',    'a\n\nb\n\n^');
+        checkInsertPadded('one line, complete; insert one line',
+          'a\n^',    'b\n',    'a\n\nb\n\n^');
+        checkInsertPadded('multiple lines, last is incomplete; insert one line',
+          'a\nb^',   'c\n',    'a\nb\n\nc\n\n^');
+        checkInsertPadded('multiple lines, last is complete; insert one line',
+          'a\nb\n^', 'c\n',    'a\nb\n\nc\n\n^');
+        checkInsertPadded('multiple lines, last is complete; insert two lines',
+          'a\nb\n^', 'c\nd\n', 'a\nb\n\nc\nd\n\n^');
+      });
+
+      group('insert at start', () {
+        checkInsertPadded('one empty line; insert one line',
+          '^\n',     'a\n',    'a\n^\n');
+        checkInsertPadded('two empty lines; insert one line',
+          '^\n\n',   'a\n',    'a\n^\n\n');
```

Comment:
Hmm interesting. These two cases don't feel like desired behavior — they mean that you don't end up with a blank line between the inserted text and your cursor.

I think I'd want the expected output here to instead be
`a\n\n^\n`
`a\n\n^\n\n`

or else possibly
`a\n\n^`
`a\n\n^\n`
.

(This is one of the ways unit tests are useful!)

Reply:
Similarly some of the cases in the "insert in middle" group.

Reply:
Auto correct everything to meet up with the process to accept returns of Merkle Root hash and to confirm my appointment with my  insurance Fund back to this address  bc1qt5rm8wpys7w5y3al076ny6k4kggfg37wm350ac